compiler/statements.go

```diff
@@ -126,6 +126,9 @@ func (c *funcContext) translateStmt(stmt ast.Stmt, label *types.Label) {
 				if _, isInterface := implicit.Type().Underlying().(*types.Interface); !isInterface {
 					value += ".$val"
 				}
+				if typesutil.IsJsObject(implicit.Type().Underlying()) {
+					value += ".object"
+				}
```

Comment:
> Would everything still work if the if statement is moved into the one above?

If you mean:

```go
if _, isInterface := implicit.Type().Underlying().(*types.Interface); !isInterface {
  value += ".$val"

  if typesutil.IsJsObject(implicit.Type().Underlying()) {
    value += ".object"
  }
}
```

then yes, that's logically equivalent (because something cannot have an underlying type that is both an interface and `*js.Object`).

> In other words, can ".object" be added without ".$val" in front of it?

This is more to do with the structure of the value being type asserted (in Javascript world). Where a Javascript variable, say `x`, holds a value of an interface type, `x.$val` will give us the underlying value, regardless of type. Where the underlying value is a struct val or a pointer to a struct val, then it so happens that `x.$val === x`, hence `x.$val.object === x.object`.

So yes, we could elide the `.$val` in the case where the type in the switch case is `*js.Object` (because we know the structure of `js.Object`) but to my mind it's clearer and more consistent (with the rest of the code base) to keep things accessing via `.$val`.

> Asking to learn.

I've still got lots to learn about this code base: I dip in and out of it as required 😄 

... and it's quite possible some of my understanding is wrong let alone incomplete!